lib/AST/ASTMangler.cpp

```diff
@@ -1228,28 +1223,54 @@ void ASTMangler::appendProtocolName(const ProtocolDecl *protocol) {
   appendDeclName(protocol);
 }
 
-void ASTMangler::appendNominalType(const NominalTypeDecl *decl) {
+void ASTMangler::appendAnyGenericType(const GenericTypeDecl *decl) {
   // Check for certain standard types.
   if (tryAppendStandardSubstitution(decl))
     return;
 
   // For generic types, this uses the unbound type.
-  TypeBase *key = decl->getDeclaredType().getPointer();
+  Type key;
+  if (auto *alias = dyn_cast<TypeAliasDecl>(decl)) {
```

Comment:
Maybe we should add a getUnboundGenericType() to TypeDecl. You already had the same bit of code in an earlier PR today, and I've written it out in a few places also.